build.csx

```diff
@@ -0,0 +1,225 @@
+        foreach (var nuspec in nuspecs)
+        {
+            Cmd(nuget, $"pack {nuspec} -Version {version} -OutputDirectory {outputDirectory} -NoPackageAnalysis");
+        }
+    });
+
+targets.Add(
+    "gitlink",
+    DependsOn("build"), () => Cmd(gitlink, $". -f {solution} -u {repoUrl} -include " + string.Join(",", gitlinks)));
+
+Run(Args, targets);
+
+// helpers
+public static void Cmd(string fileName, string args)
+{
+    using (var process = new Process())
+    {
+        process.StartInfo = new ProcessStartInfo { FileName = $"\"{fileName}\"", Arguments = args, UseShellExecute = false, };
+        Console.WriteLine($"Running '{process.StartInfo.FileName} {process.StartInfo.Arguments}'...");
+        process.Start();
+        process.WaitForExit();
+        if (process.ExitCode != 0)
+        {
+            throw new InvalidOperationException($"The command exited with code {process.ExitCode}.");
+        }
+    }
+}
+
+public static void RunMsBuild(string target)
+{
+    var packagesDirOption = string.IsNullOrEmpty(packagesDir) ? "" : $"/p:NuGetPackagesDirectory={packagesDir}";
+    Cmd(
+        msBuild,
+        $"{solution} /target:{target} /p:configuration=Release /nr:false /verbosity:minimal /nologo /fl /flp:LogFile=artifacts/logs/{target}.log;Verbosity=Detailed;PerformanceSummary {packagesDirOption}");
+}
+
+public static void RunDotNet(string workingDirectory, string command, string args="")
+{
+    using (var process = new Process())
+    {
+        process.StartInfo = new ProcessStartInfo
+        {
+            FileName = "dotnet",
+            Arguments = command + " " + args,
+            WorkingDirectory = workingDirectory,
+            UseShellExecute = false
+        };
+
+        Console.WriteLine($"Running '{process.StartInfo.FileName} {process.StartInfo.Arguments}' in '{process.StartInfo.WorkingDirectory}'...");
+        process.Start();
+        process.WaitForExit();
+        if (process.ExitCode != 0)
+        {
+            throw new InvalidOperationException($"The command exited with code {process.ExitCode}.");
+        }
+    }
+}
+
+public static void RunTests(string[] testDlls)
+{
+    foreach (var testDll in testDlls)
+    {
+        var baseFileName = Path.GetFullPath(Path.Combine(testsDirectory, Path.GetFileNameWithoutExtension(testDll))) + ".TestResults";
+        var xml = baseFileName + ".xml";
+        var html = baseFileName + ".html";
+        Cmd(xunit, $"{testDll} -noshadow -nologo -notrait \"explicit=yes\"' -xml {xml} -html {html}");
+    }
+}
+
+public static void RunDotNetTests(string[] testDirectories)
+{
+    foreach (var testDirectory in testDirectories)
+    {
+        var xml = Path.GetFullPath(Path.Combine(testsDirectory, Path.GetFileName(testDirectory) + ".TestResults.xml"));
+        RunDotNet(testDirectory, "test", $"-c Release -nologo -notrait \"explicit=yes\" -xml {xml}");
+    }
+}
+
+public static string GetVersion(string assemblyInfo)

```

Comment:
Since we only do this after build, it would be easier and cleaner to get the information from the compiled assembly, rather than from the AssemblyInfo.cs file.